includes/class-wc-gateway-amazon-payments-advanced-subscriptions.php

```diff
@@ -295,10 +295,13 @@ public function recurring_checkout_session( $payload ) {
 			);
 
 			if ( 1 === $subscriptions_in_cart ) {
+				$currency_code = get_woocommerce_currency();
+				$demicals      = 'JPY' === $currency_code ? 0 : 2;
```

Comment:
What about?

```suggestion
				$demicals      = min( wc_get_price_decimals(), 2 );
```

Not sure if it would work well tho.

Also, there's a typo on variable name `$demicals`

Reply:
Agreed @msaggiorato, much better approach. Don't even have to check currency code when implemented like that. Fixes pushed.